pkg/ruler/evaluator_remote.go

```diff
@@ -228,7 +233,7 @@ func (r *RemoteEvaluator) query(ctx context.Context, orgID, query string, ts tim
 			{Key: textproto.CanonicalMIMEHeaderKey("User-Agent"), Values: []string{userAgent}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Type"), Values: []string{mimeTypeFormPost}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Length"), Values: []string{strconv.Itoa(len(body))}},
-			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{"source=ruler"}},
+			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{queryTags}},
```

Comment:
Since the header value is a string slice, we should pack tags into a slice as well, rather than having a single string with all values.
Wdyt?

```suggestion
			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{"source=ruler", fmt.Sprintf("rule_name=%s", ruleName), fmt.Sprintf("rule_type=%s", ruleType)}},
```

Reply:
we actually expect the query tags header value to be just one comma separated string:
- https://github.com/grafana/loki/blob/main/pkg/logql/engine.go#L256
- https://github.com/grafana/loki/blob/2de6e16e19f1f011fc8b52f493a298ad750e8c64/pkg/logql/metrics.go#L566-L570